### PR TITLE
Contributions UX Improvements

### PR DESCRIPTION
**Pull request by dhaval-parekh**
Reference : Issue 1354

### CODE REVIEW

File: laterpay/asset_sources/js/blocks/contribution/edit.js [hunk 1]
```diff
@@ -238,6 +240,24 @@ class Edit extends Component {
 					{
 						'multiple' === contributionType ? (
 							<Fragment>
+								<TextControl
+									label={ __( 'Dialog Header (optional)', 'laterpay' ) }
+									placeholder={ __( 'Support the author', 'laterpay' ) }
+									value={ dialogHeader }
+									onChange={ ( newValue ) => {
+										setAttributes( { dialogHeader: newValue } )
```
Comment (thrijith) [inline laterpay/asset_sources/js/blocks/contribution/edit.js]:
```suggestion
										setAttributes( { dialogHeader: newValue } );
```
Missing semicolon

File: laterpay/asset_sources/js/blocks/contribution/edit.js [hunk 2]
```diff
@@ -238,6 +240,24 @@ class Edit extends Component {
...
+									} }
+								/>
+
+								<TextControl
+									label={ __( 'Dialog Description (optional)', 'laterpay' ) }
+									placeholder={ __( 'How much would you like to contribute?', 'laterpay' ) }
+									value={ dialogDescription }
+									onChange={ ( newValue ) => {
+										setAttributes( { dialogDescription: newValue } )
```
Comment (thrijith) [inline laterpay/asset_sources/js/blocks/contribution/edit.js]:
```suggestion
										setAttributes( { dialogDescription: newValue } );
```
Missing semicolon

File: laterpay/asset_sources/js/blocks/contribution/index.js [hunk 3]
```diff
@@ -22,6 +22,14 @@ registerBlockType( 'laterpay/contribution', {
 		campaignName: {
 			type: 'string', // Name of Campaign.
 		},
+		dialogHeader: {
+			type: 'string',
+			default: __( 'Support the author', 'laterpay' )
```
Comment (thrijith) [inline laterpay/asset_sources/js/blocks/contribution/index.js]:
```suggestion
			default: __( 'Support the author', 'laterpay' ),
```
Missing trailing comma

File: laterpay/asset_sources/js/blocks/contribution/index.js [hunk 4]
```diff
@@ -22,6 +22,14 @@ registerBlockType( 'laterpay/contribution', {
...
+		},
+		dialogDescription: {
+			type: 'string',
+			default: __( 'How much would you like to contribute?', 'laterpay' )
```
Comment (thrijith) [inline laterpay/asset_sources/js/blocks/contribution/index.js]:
```suggestion
			default: __( 'How much would you like to contribute?', 'laterpay' ),
```
Missing trailing comma

File: laterpay/views/backend/contributions.php [hunk 5]
```diff
@@ -57,6 +57,25 @@ class="lp_plugin-mode-indicator"
                             <input type="text" id="lp_thank_you_page" class="lp_input" name="contribution_thank_you_page" placeholder="http://www..." value="" />
                             <p data-icon="n" class="lp-contribution-error-message"></p>
                         </div>
+
+                        <div class="contributions_option_single hide-on-single-purchase">
+                            <label class="lp_contribution_label" for="lp_dialog_header"><?php esc_html_e( 'Dialog Header (optional)', 'laterpay' ); ?></label>
+                            <p class="lp_tooltip lp_tooltip_p lp_tooltip_contribution" data-tooltip="<?php esc_attr_e( 'Optional. Appear as Dialog box header.', 'laterpay' ); ?>">
+                                <span data-icon="m"></span>
```
Comment (thrijith) [inline laterpay/views/backend/contributions.php]: Let's remove this tooltip for now.

File: laterpay/views/backend/contributions.php [hunk 6]
```diff
@@ -57,6 +57,25 @@ class="lp_plugin-mode-indicator"
...
+                            </p><br />
+                            <input type="text" id="lp_dialog_header" class="lp_input" name="dialog_header" placeholder="<?php esc_attr_e( 'Support the author', 'laterpay' ); ?>" value=""/>
+                            <p data-icon="n" class="lp-contribution-error-message"></p>
+                        </div>
+
+                        <div class="contributions_option_single hide-on-single-purchase">
+                            <label class="lp_contribution_label" for="lp_dialog_description"><?php esc_html_e( 'Dialog Description (optional)', 'laterpay' ); ?></label>
+                            <p class="lp_tooltip lp_tooltip_p lp_tooltip_contribution" data-tooltip="<?php esc_attr_e( 'Optional. Appear as Dialog box description.', 'laterpay' ); ?>">
```
Comment (thrijith) [inline laterpay/views/backend/contributions.php]: Let's remove this tooltip for now.

File: laterpay/asset_sources/js/laterpay-backend-contributions.js [hunk 7]
```diff
@@ -58,6 +63,13 @@
                         }, 1500)
                     );
 
+                    var onchangeCallback = function () {
+                        resetGenerateButton();
+                        updateLivePreview( undefined, undefined, 'multiple' );
+                    };
+
+                    $o.dialogHeader.on( 'change', onchangeCallback );
```
Comment (thrijith) [inline laterpay/asset_sources/js/laterpay-backend-contributions.js]:
```suggestion
                    $o.dialogHeader.on( 'keyup', onchangeCallback );
```
Could we update it to `keyup` instead, merchant could see the update on preview as they type.

File: laterpay/asset_sources/js/laterpay-backend-contributions.js [hunk 8]
```diff
@@ -58,6 +63,13 @@
...
+                    $o.dialogDescription.on( 'change', onchangeCallback );
```
Comment (thrijith) [inline laterpay/asset_sources/js/laterpay-backend-contributions.js]:
```suggestion
                    $o.dialogDescription.on( 'keyup', onchangeCallback );
```
Same as above.

**Review by thrijith — CHANGES_REQUESTED** [review]

File: laterpay/asset_sources/js/blocks/contribution/edit.js (hunk 1 above)
Comment (dhaval-parekh) [reply laterpay/asset_sources/js/blocks/contribution/edit.js]: Addressed.

File: laterpay/asset_sources/js/blocks/contribution/index.js (hunk 3 above)
Comment (dhaval-parekh) [reply laterpay/asset_sources/js/blocks/contribution/index.js]: Addressed.

File: laterpay/views/backend/contributions.php (hunk 5 above)
Comment (dhaval-parekh) [reply laterpay/views/backend/contributions.php]: Addressed.

File: laterpay/asset_sources/js/laterpay-backend-contributions.js (hunk 7 above)
Comment (dhaval-parekh) [reply laterpay/asset_sources/js/laterpay-backend-contributions.js]: Addressed.

File: laterpay/asset_sources/js/laterpay-backend-contributions.js (hunk 8 above)
Comment (dhaval-parekh) [reply laterpay/asset_sources/js/laterpay-backend-contributions.js]: Addressed.

File: laterpay/asset_sources/js/blocks/contribution/index.js (hunk 4 above)
Comment (dhaval-parekh) [reply laterpay/asset_sources/js/blocks/contribution/index.js]: Addressed.

File: laterpay/asset_sources/js/blocks/contribution/edit.js (hunk 2 above)
Comment (dhaval-parekh) [reply laterpay/asset_sources/js/blocks/contribution/edit.js]: Addressed.

**Pull request closed without merging**